pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -387,7 +398,12 @@ func (c *Controller) deletePodAutoscaler(ns, name string) error {
 	return nil
 }
 
-func (c *Controller) validateAutoscaler(podAutoscaler *datadoghq.DatadogPodAutoscaler) error {
+func (c *Controller) validateAutoscaler(key string, podAutoscaler *datadoghq.DatadogPodAutoscaler) error {
+	// Check that we are within the limit of 100 DatadogPodAutoscalers
+	if !c.hashHeap.Exists(key) {
+		return fmt.Errorf("Autoscaler disabled as maximum number per cluster reached (%d)", maxDatadogPodAutoscalerObjects)
+	}
+
 	// Check that targetRef is not set to the cluster agent
 	clusterAgentPodName, err := common.GetSelfPodName()
```

Comment:
Thinking about this (not exactly related to this PR), I don't think we want to `Error` all DPA if we cannot get self pod name, we should probably just skip.